esb/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/part/EsbMultiPageEditor.java

```diff
@@ -497,37 +497,48 @@ protected void pageChange(int pageIndex) {
 		// Invoke the appropriate handler method.
 		switch (pageIndex) {
 		case DESIGN_VIEW_PAGE_INDEX: {
+			String source = sourceEditor.getDocument().get();
 			MediatorFactoryUtils.registerFactories();
-			handleDesignViewActivatedEvent(); 
-			final EsbMultiPageEditor currentEditor = this;						
-			Display.getCurrent().asyncExec(new Runnable() {
-				
-				@Override
-				public void run() {
-					EditorUtils.setLockmode(graphicalEditor, true);
-					final boolean dirty = isDirty(); //save previous status
-					//IFile file = ((IFileEditorInput)getEditorInput()).getFile();
-/*			        ElementDuplicator endPointDuplicator = new ElementDuplicator(file.getProject(),getGraphicalEditor());        
-			        endPointDuplicator.updateAssociatedDiagrams(currentEditor);*/
-
-					/**
-					 * This is to avoid RJS0007E Semantic refresh failed issue
-					 * appears in compartments, which has only one node. This
-					 * should be replaced with the better approach
-					 */
-			        AbstractEsbNodeDeserializer.relocateStartNodes();
-			        AbstractEsbNodeDeserializer.resizeParentCompartments();
-			        
-			        if(!dirty && isDirty()){
-			        	//remove any dirty status introduced by ElementDuplicator or relocateStartNodes();
-			        	 getEditor(0).doSave(new NullProgressMonitor());
-			        	 firePropertyChange(PROP_DIRTY);
-			        }
-
-			        EditorUtils.setLockmode(graphicalEditor, false);
-			        
-				}
-			});
+			try {
+				handleDesignViewActivatedEvent();
+				// final EsbMultiPageEditor currentEditor = this;
+				Display.getCurrent().asyncExec(new Runnable() {
+					@Override
+					public void run() {
+						EditorUtils.setLockmode(graphicalEditor, true);
+						final boolean dirty = isDirty(); // save previous status
+						// IFile file = ((IFileEditorInput)getEditorInput()).getFile();
+						/*
+						 * ElementDuplicator endPointDuplicator = new
+						 * ElementDuplicator(file.getProject(),getGraphicalEditor());
+						 * endPointDuplicator.updateAssociatedDiagrams(currentEditor);
+						 */
+
+						/**
+						 * This is to avoid RJS0007E Semantic refresh failed issue appears in compartments, which has
+						 * only one node. This should be replaced with the better approach
+						 */
+						AbstractEsbNodeDeserializer.relocateStartNodes();
+						AbstractEsbNodeDeserializer.resizeParentCompartments();
+
+						if (!dirty && isDirty()) {
+							// remove any dirty status introduced by ElementDuplicator or relocateStartNodes();
+							getEditor(0).doSave(new NullProgressMonitor());
+							firePropertyChange(PROP_DIRTY);
+						}
+
+						EditorUtils.setLockmode(graphicalEditor, false);
+
+					}
+				});
+			} catch (Exception e) {
```

Comment:
General Exception is thrown from Deserializer->updateDesign()
[ org.wso2.developerstudio.eclipse.gmf.esb.diagram.custom.deserializer.Deserializer.updateDesign() throws Exception ]
This was passed to here.